### PR TITLE
boards: arm: Wio-LTE-Cat.1

### PR DESCRIPTION
**Pull request by kbingham**
Provide board support for the Seeed Wio LTE Cat.1
 - http://wiki.seeedstudio.com/Wio_LTE_Cat.1/

This is an early pull-request marked as RFC, to kick off review, and try and get the tests running and of course see what else I need to do for integration.

I am aware of the lack of documentation, so I'm fully aware this isn't ready yet, that's next - but any comments on anything specific in the board support would be welcome.

Signed-off-by: Kieran Bingham <kbingham@kernel.org>

### CODE REVIEW

**Comment by MaureenHelm** [comment]
Closing and reopening to restart some of the checks (documentation, gitlint, identity/emails, and kconfig) since they seem to have gotten lost.

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1.dts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+		zephyr,ccm = &ccm0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		rgb_led_pwr: ws2812b_pwr {
+			gpios = <&gpioa 8 GPIO_INT_ACTIVE_HIGH>;
```
Comment (erwango) [inline boards/arm/wio_lte_cat_1/wio_lte_cat_1.dts]: Please rebase on top of latest master and use new GPIO API

File: boards/arm/wio_lte_cat_1/pinmux.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+static const struct pin_config pinconf[] = {
+#ifdef CONFIG_UART_1
+	{ STM32_PIN_PB6, STM32F4_PINMUX_FUNC_PB6_USART1_TX },
+	{ STM32_PIN_PB7, STM32F4_PINMUX_FUNC_PB7_USART1_RX },
+#endif  /* CONFIG_UART_1 */
+#ifdef CONFIG_UART_2
+	{ STM32_PIN_PA2, STM32F4_PINMUX_FUNC_PA2_USART2_TX },
+	{ STM32_PIN_PA3, STM32F4_PINMUX_FUNC_PA3_USART2_RX },
+#endif  /* CONFIG_UART_2 */
+#ifdef CONFIG_UART_6
```
Comment (erwango) [inline boards/arm/wio_lte_cat_1/pinmux.c]: I'm not seeing UART_6 being introduced in Kconfig or dts. Can you cross check?

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+  - gnuarmemb
+  - xtools
+ram: 192
+flash: 1024
+supported:
+  - gpio
+  - i2c
+  - counter
+  - spi
+  - usb_device
```
Comment (erwango) [inline boards/arm/wio_lte_cat_1/wio_lte_cat_1.yaml]: spi and usb_device are not enabled in dts.

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1_defconfig [hunk 4]
```diff
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_ARM=y
+CONFIG_BOARD_WIO_LTE_CAT_1=y
```
Comment (erwango) [inline boards/arm/wio_lte_cat_1/wio_lte_cat_1_defconfig]: Please remove board name

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for this new board. Few comments and a rebase needed to use new GPIO  API.

File: boards/arm/wio_lte_cat_1/Kconfig.defconfig [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+config UART_1
+	default y
+
+endif # UART_CONSOLE || SERIAL
+
+if SERIAL
+
+# QUACTEL Modem is attached to USART_2
+config UART_2
```
Comment (erwango) [inline boards/arm/wio_lte_cat_1/Kconfig.defconfig]: Please apply https://github.com/zephyrproject-rtos/zephyr/pull/22666

**Review by erwango — CHANGES_REQUESTED** [review]

File: boards/arm/wio_lte_cat_1/Kconfig.defconfig (hunk 5 above)
Comment (kbingham) [reply boards/arm/wio_lte_cat_1/Kconfig.defconfig]: Ah yes, that looks much better than lots of if/endifs.

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1_defconfig (hunk 4 above)
Comment (kbingham) [reply boards/arm/wio_lte_cat_1/wio_lte_cat_1_defconfig]: Ack.

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1.yaml (hunk 3 above)
Comment (kbingham) [reply boards/arm/wio_lte_cat_1/wio_lte_cat_1.yaml]: Will remove. USB is patches on top, but it's flaky so not included yet.

File: boards/arm/wio_lte_cat_1/pinmux.c (hunk 2 above)
Comment (kbingham) [reply boards/arm/wio_lte_cat_1/pinmux.c]: Ah, yes this is likely my debug serial. Can likely be removed and kept in my local patch.

File: boards/arm/wio_lte_cat_1/wio_lte_cat_1.dts (hunk 1 above)
Comment (kbingham) [reply boards/arm/wio_lte_cat_1/wio_lte_cat_1.dts]: Ack. I'll have to see what changed there.

**Comment by erwango** [comment]
@kbingham any plans for this change?

**Comment by kbingham** [comment]
> @kbingham any plans for this change?

Hi @erwango , Yes, I hope to get this upstreamed - but my focus has been on other projects lately I'm afraid.

Just made a big release on the $DAYJOB, so hopefully I can get some more spare time again ;-)

**Comment by erwango** [comment]
Thanks @kbingham for this feedback.
Once you're back on it, I strongly suggest to rebase and have a look to others boards to check the configuration side since there had been significant changes on this front.

**Pull request closed without merging**